packages/react-native/ReactCommon/react/performance/timeline/PerformanceEntryKeyedBuffer.cpp

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include "PerformanceEntryKeyedBuffer.h"
+#include <string>
+
+namespace facebook::react {
+
+void PerformanceEntryKeyedBuffer::add(const PerformanceEntry& entry) {
+  auto& list = entryMap_.at(entry.name);
+  auto& toConsume = toConsumeMap_.at(entry.name);
+  list.push_back(entry);
+  totalEntryCount_ += 1;
+  totalToConsume_ += 1;
+  toConsume += 1;
```

Comment:
I think we should be able to simplify this implementation, removing the concept of consuming the whole contents of the buffer. We don't have a use case for that AFAIK.

Reply:
Removed